tests/elemental3/start_k8s.pm

```diff
@@ -0,0 +1,222 @@
+# Maintainer: unified-core@suse.com, ldevulder@suse.com
+
+use base qw(opensusebasetest);
+use testapi;
+use lockapi;
+use experimental qw(switch);
+use serial_terminal qw(select_serial_terminal);
+use utils qw(exec_and_insert_password systemctl);
+use mm_network qw(configure_hostname);
+use network_utils qw(get_default_dns is_running_in_isolated_network set_resolv);
+use utils qw(file_content_replace);
+use Utils::Architectures qw(is_aarch64);
+use Mojo::File qw(path);
+
+=head2 wait_kubectl_cmd
+
+ wait_kubectl_cmd( [ timeout => <value> ] );
+
+Wait for kubectl command to be available.
+
+=cut
+
+sub wait_kubectl_cmd {
+    my %args = @_;
+    $args{timeout} //= 120;
+    my $starttime = time;
+    my $ret = undef;
+
+    while ($ret = script_run('which kubectl', ($args{timeout} / 10))) {
+        my $timerun = time - $starttime;
+        if ($timerun < $args{timeout}) {
+            sleep 5;
+        }
+        else {
+            die("kubectl command did not appear within $args{timeout} seconds!");
+        }
+    }
+
+    # Return the command status
+    die('Check did not return a defined value!') unless defined $ret;
+    return $ret;
+}
+
+=head2 wait_k8s_state
+
+ wait_k8s_state( regex => <value> [, timeout => <value> ] );
+
+Checks for up to B<$timeout> seconds whether K8s cluster is running.
+Returns 0 if cluster is running or croaks on timeout.
+
+=cut
+
+sub wait_k8s_state {
+    my %args = @_;
+    $args{timeout} //= 120;
+    my $starttime = time;
+    my $ret = undef;
+    my $chk_cmd = 'kubectl get pod -A 2>&1';
+
+    die('A regex should be defined!') if (!defined $args{regex} || $args{regex} eq '');
+
+    while (
+        $ret = script_run(
+            "! ($chk_cmd | grep -E -i -v -q '$args{regex}')",
+            ($args{timeout} / 10)
+        )
+      )
+    {
+        my $timerun = time - $starttime;
+        if ($timerun < $args{timeout}) {
+            sleep 10;
+        }
+        else {
+            record_info('K8s failed state', script_output("$chk_cmd"));
+            die("K8s cluster did not start within $args{timeout} seconds!");
+        }
+    }
+
+    # Return the command status
+    die('Check did not return a defined value!') unless defined $ret;
+    return $ret;
+}
+
+=head2 prepare_test_framework
+
+ prepare_test_framework( arch => <value>, k8s => <value> );
+
+Prepare for K8s tests to be done with ECM's distro-test-framework.
+It can be executed on different architectures and K8s distributions.
+
+=cut
+
+sub prepare_test_framework {
+    my (%args) = @_;
+
+    # Define architecture
+    my $arch;
+    given ($args{arch}) {
+        when ('aarch64') {
+            $arch = 'arm64';
+        }
+        when ('x86_64') {
+            $arch = 'amd64';
+        }
+    }
+
+    # Get some informations from the cluster
+    my ($k8s_version) = script_output('kubectl version') =~ /[Ss]erver.*:\s*(.*)/;
+    my $fqdn = script_output('hostname -f');
+    my $k8s_yaml = "/etc/rancher/$args{k8s}/$args{k8s}.yaml";
```

Comment:
Another instance of `/etc/rancher/$args{k8s}/$args{k8s}.yaml`, so I'm guessing that perhaps this is the correct file name and there's no `/etc/rancher/$args{k8s}/config.yaml` perhaps?